datafusion/src/sql/planner.rs

```diff
@@ -565,14 +587,22 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 )));
             } else {
                 let fields = plan.schema().fields().clone();
-                LogicalPlanBuilder::from(plan)
-                    .project(
+                let plan = LogicalPlanBuilder::from(plan)
+                    .project_with_alias(
                         fields
                             .iter()
                             .zip(columns_alias.iter())
-                            .map(|(field, ident)| col(field.name()).alias(&ident.value)),
+                            .map(|(field, ident)| {
+                                col_with_table_name(
+                                    field.name(),
+                                    &*(alias.clone().name.value),
+                                )
+                                .alias(ident.value.as_str())
+                            }),
```

Comment:
After taking a closer look at the code, I now agree with @alamb that we don't need to do a 2nd round of patching here. The `fields` comes from `plan.schema().fields()`. The schema for the plan we referenced here has already been patched in the previous relation match block.